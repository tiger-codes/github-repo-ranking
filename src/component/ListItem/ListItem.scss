.list-item-container {
    display: flex;
    justify-content: space-between;
    background-color: #f7fafc;
    margin: 5px;
    padding: 10px;
    .secondary {
        span {
            margin-left: 10px;
        }
    }
}
.list-item-container:hover {
    background-color: #e2f3f9;
    border: 1px solid #8fcbdf;
}