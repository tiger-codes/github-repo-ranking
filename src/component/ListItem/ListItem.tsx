import { ReactElement, useEffect, useState } from "react";
import './ListItem.scss';

interface ListItemProps {
    primaryText: string;
    secondaryText: number;
}

const ListItem = ({ primaryText, secondaryText }: ListItemProps): ReactElement => {
    return <div key={primaryText} className="list-item-container">
                <div key="primary" className="primary">{primaryText}</div>
                <div key="secondary" className="secondary">
                    <span className="fa fa-star checked"></span>
                    <span >{secondaryText}</span>
                </div>
           </div>
}

export default ListItem;
