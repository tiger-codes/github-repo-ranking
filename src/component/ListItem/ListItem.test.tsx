import * as React from 'react';
import * as renderer from 'react-test-renderer';
import ListItem from './ListItem';

describe('<ListItem />', () => {
  it('renders correctly', () => {
    const NoteComponent = renderer.create(<ListItem primaryText='abc' secondaryText={1} />).toJSON();
    expect(NoteComponent).toMatchSnapshot();
  });
});
