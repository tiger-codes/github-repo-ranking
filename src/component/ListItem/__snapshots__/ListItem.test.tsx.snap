// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> renders correctly 1`] = `
<div
  className="list-item-container"
>
  <div
    className="primary"
  >
    abc
  </div>
  <div
    className="secondary"
  >
    <span
      className="fa fa-star checked"
    />
    <span>
      1
    </span>
  </div>
</div>
`;
